websites/H/Hoyolab/presence.ts

```diff
@@ -0,0 +1,110 @@
+      data.details = "Viewing tavern page";
+      data.state = "New";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1") && document.location.search.substr(6) == "reply"){
+      data.details = "Viewing tavern page";
+      data.state = "New replies";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1") && document.location.search.substr(6) == "2"){
+      data.details = "Viewing tavern page";
+      data.state = "Featured";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/1")){
+      data.details = "Viewing tavern page";
+      data.state = "Hot";
+      data.startTimestamp = browsingStamp;
+    }
+    //Official Page
+    else if(pathname.endsWith("/home/3") && document.location.search.substr(6) == "1"){
+      data.details = "Viewing official page";
+      data.state = "Notices";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/3") && document.location.search.substr(6) == "3"){
+      data.details = "Viewing official page";
+      data.state = "Info";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/3") && document.location.search.substr(6) == "contribution"){
+      data.details = "Viewing official page";
+      data.state = "Submission events";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/3")){
+      data.details = "Viewing official page";
+      data.state = "Events";
+      data.startTimestamp = browsingStamp;
+    }
+    //Billboards
+    else if(pathname.endsWith("/home/2") && document.location.search.substr(6) == "create"){
+      data.details = "Viewing billboards page";
+      data.state = "New";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/2") && document.location.search.substr(6) == "reply"){
+      data.details = "Viewing billboards page";
+      data.state = "New replies";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/2") && document.location.search.substr(6) == "2"){
+      data.details = "Viewing billboards page";
+      data.state = "Featured";
+      data.startTimestamp = browsingStamp;
+    }
+    else if(pathname.endsWith("/home/2")){
+      data.details = "Viewing billboards page";
+      data.state = "Hot";
+      data.startTimestamp = browsingStamp;
+    }
+    //Articles
+    else if(pathname.startsWith("/genshin/article/")){
+      const title = document.querySelector(".mhy-article-page__title > h1").textContent,
+      author = document.querySelector(".mhy-article-page-author > .mhy-user-card__info > a > span").innerHTML,
+      link = window.location.href;
+
+      data.details = title;
+      data.state = "by: " + author;
+      data.startTimestamp = browsingStamp;
+      data.buttons = [{ label: "Visit", url: link }];
```

Comment:
Visit, probably ``Visit Article`` would be better

Reply:
Yeah, I'm never good with those button labels xD